### PR TITLE
New recipe: GlibNetworking

### PR DESCRIPTION
**Pull request by lukebemish**
Here's my attempt to get GlibNetworking bundled - at least the OpenSSL based bit. I am able to get libsoup to build based off of this locally, at least, so I could open a PR for that too. Supersedes #5689.

### CODE REVIEW

**Comment by lukebemish** [comment]
Hmm, I think I left some of my local debugging code in that. Let me fix that

**Comment by lukebemish** [comment]
... Actually, I have no clue why that isn't building; is `should_build_platform` obsolete now?

**Comment by lukebemish** [comment]
Converting this to a draft for now; there's some disparity between what I'm doing locally and what is happening here that I need to look into

**Comment by lukebemish** [comment]
Alright; I suppose that's what I get for testing the script on its own, not in the repo structure.

File: G/GlibNetworking/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+products_windows = [
+    FileProduct("bin/libgioenvironmentproxy.dll", :libgioenvironmentproxy),
+    FileProduct("bin/libgioopenssl.dll", :libgioopenssl)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    HostBuildDependency("Gettext_jll")
+    Dependency("OpenSSL_jll", v"1.1.20+0"; compat="~1.1.20")
```
Comment (giordano) [inline G/GlibNetworking/build_tarballs.jl:61]:
```suggestion
    Dependency("OpenSSL_jll"; compat="1.1.10")
```

File: G/GlibNetworking/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    Dependency("Glib_jll", v"2.74.0"; compat="~2.74.0")
```
Comment (giordano) [inline G/GlibNetworking/build_tarballs.jl:62]: I don't see why restricting compat to the minor version
```suggestion
    Dependency("Glib_jll"; compat="2.74.0")
```

File: G/GlibNetworking/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+MESON_FLAGS+=(--buildtype=release)
+MESON_FLAGS+=(-Dopenssl=enabled)
+
+meson "${MESON_FLAGS[@]}" glib-networking-*/
+ninja -j${nproc} --verbose
+ninja install
+
+
+# Remove temporary links
+rm ${prefix}/bin/glib-compile-{resources,schemas}
```
Comment (giordano) [inline G/GlibNetworking/build_tarballs.jl:38]:
```suggestion
rm ${bindir}/glib-compile-{resources,schemas}
```

**Pull request merged**